lib/mariaex/geometry/point.ex

```diff
@@ -0,0 +1,44 @@
+defmodule Mariaex.Geometry.Point do
+
+  @moduledoc """
+  Defines the Point struct. Implements the Ecto.Type behaviour
+  """
+
+  @type t :: %Mariaex.Geometry.Point{ coordinates: {number, number}, srid: integer }
+  defstruct coordinates: {0, 0}, srid: nil
+
+  if Code.ensure_loaded?(Ecto.Type) do
+    @behaviour Ecto.Type
+
+    def type, do: :geometry
+
+    def blank?(_), do: false
+
+    def load(%__MODULE__{} = point), do: {:ok, point}
+    def load(_), do: :error
+
+    def dump(%__MODULE__{} = point), do: {:ok, point}
+    def dump(_), do: :error
+
+    def cast(%__MODULE__{} = point), do: {:ok, point}
+    def cast(_), do: :error
+  end
+
+  @type point :: Mariaex.Geometry.Point.t
+
+  @doc """
+  Takes a Point and returns a WKB binary for MySQL
+  """
+  @spec encode(point) :: binary
+  def encode(%__MODULE__{coordinates: {x, y}, srid: srid}) do
+    srid = case srid do
+      srid when is_integer(srid) -> << srid::little-32 >>
+      _ -> << 0::little-32 >>
```

Comment:
From the MySQL docs:

> 
In MySQL, all computations are done assuming SRID 0, regardless of the actual SRID value. SRID 0 represents an infinite flat Cartesian plane with no units assigned to its axes. In the future, computations may use the specified SRID values. To ensure SRID 0 behavior, create geometry values using SRID 0. SRID 0 is the default for new geometry values if no SRID is specified.